src/components/ValidatedInput/ValidFieldPassword.tsx

```diff
@@ -124,28 +119,64 @@ export class ValidFieldPassword extends React.Component<ValidFieldPasswordProps,
   };
 
   handleGenerateText = () => {
-    const { hasGeneratePassword, strengthLevelIndicator, customField } = this.props;
-    const { levelColor } = this.state;
+    const { hasGeneratePassword, customField, hasPasswordStrenghtMeter } = this.props;
+    const { passwordStrenghtLevel } = this.state;
     return (
       <>
+        {hasPasswordStrenghtMeter && passwordStrenghtLevel && (
+          <div className="password-strength-meter">
+            <div className={classNames("strength-meter-bar", passwordStrenghtLevel)} />
+            <span className={classNames("strength-meter-label")}>{translation("password.strength.label")}</span>
+            <span className={classNames("strength-meter-level", passwordStrenghtLevel)}>
+              {translation(`password.strength.level.${passwordStrenghtLevel}`)}
+            </span>
+          </div>
+        )}
         {hasGeneratePassword && (
           <Button appearance="link" spacing="none" className="generate-password" onClick={this.handleGeneratePassword}>
             {translation("password.generateNew")}
           </Button>
         )}
-        {strengthLevelIndicator && (
-          <span className={classNames("level", levelColor)}>{translation(`password.${strengthLevelIndicator}`)}</span>
-        )}
         {customField}
       </>
     );
   };
 
+  calculatePasswordStrenght = (password: string) => {
```

Comment:
typo here. should be `strength` not `strenght`

Reply:
done